src/library_src/PredicateTypes.cpp

```diff
@@ -0,0 +1,59 @@
+#include "PredicateTypes.hpp"
+
+
+Predicate_1or2and3::Predicate_1or2and3(std::function<void()> callback, const TriggerType &or1Operand,
+                                       const TriggerType &or2Operand, const TriggerType &andOperand)
+        : Predicate(callback), or1Operand(or1Operand), or2Operand(or2Operand), andOperand(andOperand)
+{
+    for (auto &&trigger : getPredicateOperands()) {
+        triggerState[trigger] = false;
+    }
+}
+
+std::vector<TriggerType> Predicate_1or2and3::getPredicateOperands() const {
+    return {or1Operand, or2Operand, andOperand};
+}
+
+bool Predicate_1or2and3::isPredicateSatisfied() const {
+
+    if (triggerState.at(or1Operand) && triggerState.at(or2Operand) && lastSatisfied && triggerState.at(andOperand)) {
```

Comment:
shouldn't be if((triggerState.at(or1Operand) || triggerState.at(or2Operand)) && lastSatisfied && triggerState.at(andOperand)    <- or instead of and between first 2 operands?

Reply:
Nope. It checks the case after one of the operand already fired. It is kindof oposite to the normal statement (already drown the truth tables)